projects/views.py

```diff
@@ -2,83 +2,34 @@
 from django.shortcuts import get_object_or_404
 from rest_framework import generics, status
 from rest_framework.response import Response
-from rest_framework.views import APIView
 
 from components.filters import ComponentFilter
 from components.models import Component
 from components.serializers import ComponentListBasicSerializer
 
-from .models import Project
-from .serializers import (
-    ProjectControlSerializer,
-    ProjectListSerializer,
-    ProjectSerializer,
-)
+from blueprintapi.filters import ObjectPermissionsFilter
+from projects.models import Project
+from projects.serializers import ProjectControlSerializer, ProjectListSerializer, ProjectSerializer
 
 
 class ProjectsListViews(generics.ListCreateAPIView):
-
     queryset = Project.objects.all().order_by("pk")
     serializer_class = ProjectListSerializer
+    filter_backends = [ObjectPermissionsFilter, ]
 
-    def list(self, request, *args, **kwargs):
-        projects = self.get_queryset()
-        serializer = ProjectListSerializer(projects, many=True)
-        return Response(serializer.data, status=status.HTTP_200_OK)
-
-
-class ProjectsDetailView(APIView):
-    def get_object(self, project_id):
-        try:
-            return Project.objects.get(id=project_id)
-        except Project.DoesNotExist:
-            return None
 
-    def get(self, request, project_id, *args, **kwargs):
-        project_instance = get_object_or_404(Project, pk=project_id)
-        serializer = ProjectSerializer(project_instance)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+class ProjectsDetailView(generics.RetrieveUpdateDestroyAPIView):
```

Comment:
Love that using the generics here cleans up the code and brings consistency in implementation!